src/otx/core/model/module/base.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Class definition for base lightning module used in OTX."""
+from __future__ import annotations
+
+from typing import Any
+
+import torch
+from lightning import LightningModule
+from torch import Tensor
+
+from otx.core.data.entity.base import OTXBatchDataEntity
+from otx.core.model.entity.base import OTXModel
+
+
+class OTXLitModule(LightningModule):
+    """Base class for the lightning module used in OTX."""
+
+    def __init__(
+        self,
+        otx_model: OTXModel,
+        optimizer: torch.optim.Optimizer,
+        scheduler: torch.optim.lr_scheduler.LRScheduler,
+        torch_compile: bool,
+    ):
+        super().__init__()
+
+        self.model = otx_model
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.torch_compile = torch_compile
+
+        # this line allows to access init params with 'self.hparams' attribute
+        # also ensures init params will be stored in ckpt
+        self.save_hyperparameters(logger=False, ignore=["otx_model"])
+
+    def training_step(self, inputs: OTXBatchDataEntity, batch_idx: int) -> Tensor:
+        """Step for model training."""
+        train_loss = self.model(inputs)
+
+        if isinstance(train_loss, Tensor):
+            self.log(
+                "train/loss",
+                train_loss,
+                on_step=True,
+                on_epoch=False,
+                prog_bar=True,
+            )
+            return train_loss
+        if isinstance(train_loss, dict):
+            for k, v in train_loss.items():
+                self.log(
+                    f"train/{k}",
+                    v,
+                    on_step=True,
+                    on_epoch=False,
+                    prog_bar=True,
+                )
+
+            train_loss = sum(train_loss.values())
```

Comment:
Is it okay to change the variable's value? It seems that we need to use other variable name in this case. 
```python3
total_train_loss = sum(train_loss.values())
self.log(
    "train/total_loss",
    "total_train_loss",
    ...
)
```

Reply:
Is it blocking your work for now?